src/domain/relayer/mod.rs

```diff
@@ -476,6 +502,90 @@ impl<
+                let network_repo = state
+                    .network_repository()
+                    .get_by_name(NetworkType::Midnight, &relayer.network)
+                    .await
+                    .ok()
+                    .flatten()
+                    .ok_or_else(|| {
+                        RelayerError::NetworkConfiguration(format!(
+                            "Network {} not found",
+                            relayer.network
+                        ))
+                    })?;
+
+                let network = MidnightNetwork::try_from(network_repo)?;
+                let indexer_urls = network.indexer_urls.clone();
+                let metadata = HashMap::from([
+                    (
+                        "network".to_string(),
+                        format!("{:?}", to_midnight_network_id(&relayer.network)),
+                    ),
+                    ("http".to_string(), indexer_urls.http),
+                    ("ws".to_string(), indexer_urls.ws),
+                ]);
+
+                let midnight_provider: MidnightProvider = get_network_provider(
+                    &network,
+                    relayer.custom_rpc_urls.clone(),
+                    Some(&metadata),
+                )?;
+
+                // Convert network to NetworkId for sync manager
+                let network_id = to_midnight_network_id(&relayer.network);
+
+                // Create the Midnight signer to get the wallet seed
+                let midnight_signer =
+                    MidnightSignerFactory::create_midnight_signer(&signer.into(), network_id)
+                        .map_err(|e| {
+                            RelayerError::NetworkConfiguration(format!(
+                                "Failed to create signer: {}",
+                                e
+                            ))
+                        })?;
+                let wallet_seed = midnight_signer.wallet_seed();
+
+                // Get the indexer client from provider
+                let indexer_client = midnight_provider.get_indexer_client();
+
+                // Create the sync manager
+                let sync_manager = SyncManager::<QuickSyncStrategy, RSR>::new(
+                    indexer_client,
+                    wallet_seed,
+                    network_id,
+                    state.sync_state_store(),
+                    relayer.id.clone(),
+                )
+                .await
+                .map_err(|e| {
+                    RelayerError::NetworkConfiguration(format!(
+                        "Failed to create sync manager: {}",
+                        e
+                    ))
+                })?;
+
+                let transaction_counter_service = Arc::new(TransactionCounterService::new(
+                    relayer.id.clone(),
+                    relayer.address.clone(),
+                    state.transaction_counter_store(),
+                ));
+
+                let dependencies = MidnightRelayerDependencies::new(
```

Comment:
Seems like MidnightRelayerDependencies is just used temporarily, to construct it here and to pass to ::new methods where its params are extract and stored in relayer instance.

Not sure how much it helps/simplifies.